crates/fiber/src/arch/aarch64.S

```diff
@@ -10,12 +10,29 @@
 
 #include "header.h"
 
+#ifndef CFG_TARGET_OS_macos
+    .pushsection .note.gnu.property, "a";
+    .p2align 3;
+    .word 4;
+    .word 16;
+    .word 5;
+    .asciz "GNU";
+    .word 0xc0000000; // GNU_PROPERTY_AARCH64_FEATURE_1_AND
+    .word 4;
+    .word 3; // GNU_PROPERTY_AARCH64_FEATURE_1_BTI | GNU_PROPERTY_AARCH64_FEATURE_1_PAC
+    .word 0;
+    .popsection
+#endif
```

Comment:
Personally I'm pretty unfamiliar with all of this. Would you be up for adding a comment here to why this is necessary and perhaps an online doc if possible explaining the format of this section?